sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtilTest.java

```diff
@@ -58,4 +71,219 @@ void getOtlpProtocolDefault() {
+    assertThatCode(
+            configureEndpointCallable(
+                ImmutableMap.of(GENERIC_ENDPOINT_KEY, "http://localhost:4317")))
+        .doesNotThrowAnyException();
+    assertThatCode(
+            configureEndpointCallable(
+                ImmutableMap.of(GENERIC_ENDPOINT_KEY, "http://localhost:4317/")))
+        .doesNotThrowAnyException();
+    assertThatCode(
+            configureEndpointCallable(ImmutableMap.of(GENERIC_ENDPOINT_KEY, "http://localhost")))
+        .doesNotThrowAnyException();
+    assertThatCode(
+            configureEndpointCallable(ImmutableMap.of(GENERIC_ENDPOINT_KEY, "https://localhost")))
+        .doesNotThrowAnyException();
+    assertThatCode(
+            configureEndpointCallable(
+                ImmutableMap.of(GENERIC_ENDPOINT_KEY, "http://foo:bar@localhost")))
+        .doesNotThrowAnyException();
+    assertThatCode(
+            configureEndpointCallable(
+                ImmutableMap.of(
+                    GENERIC_ENDPOINT_KEY,
+                    "http://localhost:4317/path",
+                    "otel.exporter.otlp.protocol",
+                    "http/protobuf")))
+        .doesNotThrowAnyException();
+  }
+
+  @Test
+  void configureOtlpExporterBuilder_InvalidEndpoints() {
+    assertThatThrownBy(configureEndpointCallable(ImmutableMap.of(GENERIC_ENDPOINT_KEY, "/foo/bar")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must be a valid URL:");
+    assertThatThrownBy(
+            configureEndpointCallable(
+                ImmutableMap.of(GENERIC_ENDPOINT_KEY, "file://localhost:4317")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint scheme must be http or https:");
+    assertThatThrownBy(
+            configureEndpointCallable(
+                ImmutableMap.of(GENERIC_ENDPOINT_KEY, "http://localhost:4317?foo=bar")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a query string:");
+    assertThatThrownBy(
+            configureEndpointCallable(
+                ImmutableMap.of(GENERIC_ENDPOINT_KEY, "http://localhost:4317#fragment")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a fragment:");
+    assertThatThrownBy(
+            configureEndpointCallable(
+                ImmutableMap.of(
+                    GENERIC_ENDPOINT_KEY,
+                    "http://localhost:4317/path",
+                    "otel.exporter.otlp.protocol",
+                    "grpc")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a path:");
+  }
+
+  private static ThrowingCallable configureEndpointCallable(Map<String, String> properties) {
+    return () -> configureEndpoint(DATA_TYPE_TRACES, properties);
+  }
+
+  @Test
+  void configureOtlpExporterBuilder_HttpGenericEndpointKey() {
```

Comment:
I've tried to improve the naming conventions in the tests. There are now 3 tests:
- `configureOtlpExporterBuilder_HttpGenericEndpointKey` validates the behavior when the generic `otel.exporter.otlp.endpoint` is used to configure the http endpoint
- `configureOtlpExporterBuilder_HttpTracesEndpointKey` validates the behavior when the trace specific `otel.exporter.otlp.traces.endpoint` is used to configure the http endpoint (included validating that it overrides the generic endpoint value)
- `configureOtlpExporterBuilder_HttpMetricsEndpointKey` validates the behavior when the metrics specific `otel.exporter.otlp.metrics.endpoint` is used to configure the http endpoint (included validating that it overrides the generic endpoint value)

I think the method names are pretty good. In any case, they're better than repeating a ~7 line `ImmutableMap.of(..)` 20+ times. If its not immediately obvious what the `configureEndpointForHttp` does, take a quick look at it - IMO it greatly improves the readability of the test cases.